### PR TITLE
Added hoist and proptypes

### PR DESCRIPTION
**Pull request by iaronaraujo**
#### What does this PR do? \*

Adds `hoistNonReactStatic` and `propTypes` to `withCssHandles`

### CODE REVIEW

File: react/withCssHandles.tsx [hunk 1]
```diff
@@ -16,7 +17,8 @@ const withCssHandles = <T extends CssHandlesInput, ComponentProps>(
 
   const displayName = Component.displayName || Component.name || 'Component'
   EnhancedComponent.displayName = `withCssHandles(${displayName})`
-
+  hoistNonReactStatic(EnhancedComponent, Component)
+  EnhancedComponent.propTypes = Component.propTypes
```
Comment (klzns) [inline react/withCssHandles.tsx]: Add defaultProps as well

File: react/withCssHandles.tsx [hunk 2]
```diff
@@ -16,7 +17,9 @@ const withCssHandles = <T extends CssHandlesInput, ComponentProps>(
 
   const displayName = Component.displayName || Component.name || 'Component'
   EnhancedComponent.displayName = `withCssHandles(${displayName})`
-
+  hoistNonReactStatic(EnhancedComponent, Component)
```
Comment (lbebber) [inline react/withCssHandles.tsx]: I think this should be done
```suggestion
  return hoistNonReactStatic(EnhancedComponent, Component)
```

Comment (lbebber) [reply react/withCssHandles.tsx]: I might be mistaken though, please confirm

**Comment by iaronaraujo** [comment]
Closed because we decided that a component proptype should not be used by another component. SO if you have this issue with an app, you should fix this on the app, not on the css-handles

**Pull request closed without merging**